thrust/thrust/detail/config/namespace.h

```diff
@@ -84,6 +87,84 @@
 #define THRUST_NS_QUALIFIER ::thrust
 #endif
 
+// clang-format off
+#if THRUST_DEVICE_SYSTEM == THRUST_DEVICE_SYSTEM_CUDA
```

Comment:
question (maybe blocking): Can people still use the CUDA backend even if it's not configured as the default device system? If so, then they would still run into the same issues.

Reply:
There are many CUDA-related things hidden under a check like that. For instance, temporary storage doesn't compile with CUDA backend when the device system is not CUDA: https://github.com/NVIDIA/cccl/blob/ea33ba965713ae647dd9de4814260fea345bee9e/thrust/thrust/detail/allocator/temporary_allocator.inl#L53-L59
The CUDA system also can't be used as host system. In other words, I don't know how people would use `thrust::cuda::par` directly without `THRUST_DEVICE_SYSTEM_CUDA`.